### ISSUE TITLE
SPC-. not fails with error

### ISSUE THREAD

**Issue by shanavas786** [hlissner/doom-emacs#1234]
### Observed behavior

Lots of key sequences fails with `Key sequence M-SPC h m starts with non-prefix key M-SPC`
![doom](https://user-images.githubusercontent.com/5337197/54041532-f67db980-41ed-11e9-87f7-ed9a900736ac.gif)

`DEBUG=1 make doctor` shows the error below
```
  > (:lang rust) Syntax error: (error Key sequence M-SPC h m starts
    with non-prefix key M-SPC)
```

### Steps to reproduce

1. start emacs
2. press `SPC-.`

### System information

<details>
<summary>Click to expand</summary>

```
Doom Doctor
Emacs v27.0.50
Doom v2.0.9 (1f23eecd)
shell: /bin/bash
Compiled with:
  XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY
  INOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB
  TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM THREADS LIBSYSTEMD PDUMPER LCMS2 GMP
uname -a:
  Linux sinergia 4.19.0-2-amd64 #1 SMP Debian 4.19.16-1 (2019-01-17) x86_64
  GNU/Linux

Checking your OS...
Checking your fonts...
Checking gnutls/openssl...
Testing your root certificates...
Checking for GNU/BSD tar...
DOOM Running doom hook: evil-mode
DOOM Running doom hook: +evil|init-popup-rules
DOOM Running doom hook: +workspaces|init
DOOM Initialized evil-collection-info [feature/evil]
DOOM Initialized evil-collection-comint [feature/evil]
DOOM Initialized evil-collection-custom [feature/evil]
DOOM Initialized evil-collection-help [feature/evil]
DOOM Registered package ’xclip’
DOOM Registered package ’exec-path-from-shell’
DOOM Registered package ’osx-clipboard’
DOOM Registered package ’ns-auto-titlebar’
DOOM Registered package ’all-the-icons’
DOOM Registered package ’hide-mode-line’
DOOM Registered package ’highlight-indentation’
DOOM Registered package ’highlight-numbers’
DOOM Registered package ’highlight-escape-sequences’ with recipe (:fetcher github :repo hlissner/highlight-escape-sequences)
DOOM Registered package ’nlinum’
DOOM Registered package ’nlinum-hl’
DOOM Registered package ’nlinum-relative’
DOOM Registered package ’rainbow-delimiters’
DOOM Registered package ’visual-fill-column’
DOOM Registered package ’restart-emacs’
DOOM Registered package ’ace-link’
DOOM Registered package ’ace-window’
DOOM Registered package ’avy’
DOOM Registered package ’command-log-mode’
DOOM Registered package ’dtrt-indent’
DOOM Registered package ’helpful’
DOOM Registered package ’pcre2el’
DOOM Registered package ’smartparens’
DOOM Registered package ’undo-tree’
DOOM Registered package ’ws-butler’
DOOM Registered package ’projectile’
DOOM Registered package ’general’
DOOM Registered package ’which-key’
DOOM Registered package ’hydra’
DOOM Registered package ’esup’
DOOM Registered package ’buttercup’
DOOM Registered package ’emacs-snippets’ with recipe (:fetcher github :repo hlissner/emacs-snippets :files (*))
DOOM Registered package ’quickrun’
DOOM Registered package ’evil’
DOOM Registered package ’evil-args’
DOOM Registered package ’evil-commentary’
DOOM Registered package ’evil-easymotion’
DOOM Registered package ’evil-embrace’
DOOM Registered package ’evil-escape’
DOOM Registered package ’evil-exchange’
DOOM Registered package ’evil-indent-plus’
DOOM Registered package ’evil-matchit’
DOOM Registered package ’evil-numbers’
DOOM Registered package ’evil-textobj-anyblock’
DOOM Registered package ’evil-snipe’
DOOM Registered package ’evil-surround’
DOOM Registered package ’evil-visualstar’
DOOM Registered package ’exato’
DOOM Registered package ’neotree’
DOOM Registered package ’evil-collection’
DOOM Registered package ’yasnippet’
DOOM Registered package ’helm’
DOOM Registered package ’dumb-jump’
DOOM Registered package ’ivy-xref’
DOOM Registered package ’helm-xref’
DOOM Registered package ’helm-dash’
DOOM Registered package ’counsel-dash’
DOOM Registered package ’yasnippet’
DOOM Registered package ’auto-yasnippet’
DOOM Registered package ’emacs-snippets’ with recipe (:fetcher github :repo hlissner/emacs-snippets :files (*))
DOOM Registered package ’persp-mode’
DOOM Registered package ’company’
DOOM Registered package ’company-dict’
DOOM Registered package ’company-prescient’
DOOM Registered package ’company-box’
DOOM Registered package ’helm’
DOOM Registered package ’helm-ag’
DOOM Registered package ’helm-c-yasnippet’
DOOM Registered package ’helm-company’
DOOM Registered package ’helm-describe-modes’ with recipe (:fetcher github :repo emacs-helm/helm-describe-modes)
DOOM Registered package ’helm-projectile’
DOOM Registered package ’swiper-helm’
DOOM Registered package ’helm-flx’
DOOM Registered package ’posframe’
DOOM Registered package ’doom-themes’
DOOM Registered package ’solaire-mode’
DOOM Registered package ’doom-modeline’
DOOM Registered package ’anzu’
DOOM Registered package ’evil-anzu’
DOOM Registered package ’evil-goggles’
DOOM Registered package ’fill-column-indicator’
DOOM Registered package ’hl-todo’
DOOM Registered package ’nav-flash’
DOOM Registered package ’treemacs’
DOOM Registered package ’treemacs-evil’
DOOM Registered package ’treemacs-projectile’
DOOM Registered package ’unicode-fonts’
DOOM Registered package ’git-gutter-fringe’
DOOM Registered package ’vi-tilde-fringe’
DOOM Registered package ’switch-window’
DOOM Registered package ’ace-window’
DOOM Registered package ’format-all’
DOOM Registered package ’evil-multiedit’
DOOM Registered package ’evil-mc’
DOOM Registered package ’multiple-cursors’
DOOM Registered package ’rotate-text’ with recipe (:fetcher github :repo debug-ito/rotate-text.el)
DOOM Registered package ’evil-vimish-fold’
DOOM Registered package ’dired-k’
DOOM Registered package ’ranger’
DOOM Registered package ’all-the-icons-dired’
DOOM Registered package ’imenu-anywhere’
DOOM Registered package ’imenu-list’
DOOM Registered package ’git-link’
DOOM Registered package ’git-timemachine’
DOOM Registered package ’gitconfig-mode’
DOOM Registered package ’gitignore-mode’
DOOM Registered package ’flyspell-correct’
DOOM Registered package ’flyspell-correct-ivy’
DOOM Registered package ’flyspell-correct-helm’
DOOM Registered package ’flyspell-correct-popup’
DOOM Registered package ’flycheck’
DOOM Registered package ’flycheck-popup-tip’
DOOM Registered package ’flycheck-posframe’
DOOM Registered package ’editorconfig’
DOOM Registered package ’makefile-executor’
DOOM Registered package ’magit’
DOOM Registered package ’forge’
DOOM Registered package ’magit-gitflow’
DOOM Registered package ’magit-todos’
DOOM Registered package ’evil-magit’
DOOM Registered package ’cmake-mode’
DOOM Registered package ’cuda-mode’
DOOM Registered package ’demangle-mode’
DOOM Registered package ’disaster’
DOOM Registered package ’modern-cpp-font-lock’
DOOM Registered package ’opencl-mode’
DOOM Registered package ’glsl-mode’
DOOM Registered package ’company-glsl’ with recipe (:fetcher github :repo Kaali/company-glsl)
DOOM Registered package ’ccls’
DOOM Registered package ’irony’
DOOM Registered package ’irony-eldoc’
DOOM Registered package ’flycheck-irony’
DOOM Registered package ’company-irony’
DOOM Registered package ’company-irony-c-headers’
DOOM Registered package ’rtags’
DOOM Registered package ’ivy-rtags’
DOOM Registered package ’helm-rtags’
DOOM Registered package ’graphql-mode’
DOOM Registered package ’json-mode’
DOOM Registered package ’toml-mode’
DOOM Registered package ’yaml-mode’
DOOM Registered package ’csv-mode’
DOOM Registered package ’dhall-mode’
DOOM Registered package ’auto-compile’
DOOM Registered package ’highlight-quoted’
DOOM Registered package ’macrostep’
DOOM Registered package ’overseer’
DOOM Registered package ’elisp-def’
DOOM Registered package ’flycheck-cask’
DOOM Registered package ’coffee-mode’
DOOM Registered package ’js2-mode’
DOOM Registered package ’rjsx-mode’
DOOM Registered package ’typescript-mode’
DOOM Registered package ’eslintd-fix’
DOOM Registered package ’js2-refactor’
DOOM Registered package ’nodejs-repl’
DOOM Registered package ’npm-mode’
DOOM Registered package ’skewer-mode’
DOOM Registered package ’xref-js2’
DOOM Registered package ’tide’
DOOM Registered package ’markdown-mode’
DOOM Registered package ’markdown-toc’
DOOM Registered package ’pandoc-mode’
DOOM Registered package ’org-plus-contrib’
DOOM Registered package ’org’
DOOM Registered package ’org-bullets’ with recipe (:fetcher github :repo Kaligule/org-bullets)
DOOM Registered package ’org-yt’ with recipe (:fetcher github :repo TobiasZawada/org-yt)
DOOM Registered package ’toc-org’
DOOM Registered package ’evil-org’
DOOM Registered package ’org-pdfview’
DOOM Registered package ’org-download’
DOOM Registered package ’ob-async’
DOOM Registered package ’ob-mongo’
DOOM Registered package ’ob-sql-mode’
DOOM Registered package ’ob-translate’
DOOM Registered package ’ob-ipython’
DOOM Registered package ’ob-crystal’
DOOM Registered package ’ob-go’
DOOM Registered package ’ob-nim’
DOOM Registered package ’ob-racket’ with recipe (:fetcher github :repo DEADB17/ob-racket)
DOOM Registered package ’ob-restclient’
DOOM Registered package ’ob-rust’
DOOM Registered package ’ox-pandoc’
DOOM Registered package ’htmlize’
DOOM Registered package ’centered-window’ with recipe (:fetcher github :repo anler/centered-window-mode)
DOOM Registered package ’org-tree-slide’
DOOM Registered package ’ox-reveal’
DOOM Registered package ’nose’
DOOM Registered package ’python-pytest’
DOOM Registered package ’pip-requirements’
DOOM Registered package ’pipenv’
DOOM Registered package ’pyvenv’
DOOM Registered package ’pyenv-mode’
DOOM Registered package ’conda’
DOOM Registered package ’anaconda-mode’
DOOM Registered package ’company-anaconda’
DOOM Registered package ’rust-mode’
DOOM Registered package ’flycheck-rust’
DOOM Registered package ’racer’
DOOM Registered package ’sbt-mode’
DOOM Registered package ’scala-mode’
DOOM Registered package ’lsp-scala’
DOOM Registered package ’ensime’
DOOM Registered package ’company-shell’
DOOM Registered package ’fish-mode’
DOOM Registered package ’circe’
DOOM Registered package ’circe-notifications’
DOOM Registered package ’emacs-snippets’ with recipe (:fetcher github :repo hlissner/emacs-snippets :files (*))
Checking your enabled modules...
DOOM Registered package ’quickrun’
DOOM Registered package ’evil’
DOOM Registered package ’evil-args’
DOOM Registered package ’evil-commentary’
DOOM Registered package ’evil-easymotion’
DOOM Registered package ’evil-embrace’
DOOM Registered package ’evil-escape’
DOOM Registered package ’evil-exchange’
DOOM Registered package ’evil-indent-plus’
DOOM Registered package ’evil-matchit’
DOOM Registered package ’evil-numbers’
DOOM Registered package ’evil-textobj-anyblock’
DOOM Registered package ’evil-snipe’
DOOM Registered package ’evil-surround’
DOOM Registered package ’evil-visualstar’
DOOM Registered package ’exato’
DOOM Registered package ’neotree’
DOOM Registered package ’evil-collection’
DOOM Registered package ’yasnippet’
DOOM Registered package ’helm’
DOOM Registered package ’dumb-jump’
DOOM Registered package ’ivy-xref’
DOOM Registered package ’helm-xref’
DOOM Registered package ’helm-dash’
DOOM Registered package ’counsel-dash’
DOOM Registered package ’yasnippet’
DOOM Registered package ’auto-yasnippet’
DOOM Registered package ’emacs-snippets’ with recipe (:fetcher github :repo hlissner/emacs-snippets :files (*))
DOOM Registered package ’persp-mode’
DOOM Registered package ’company’
DOOM Registered package ’company-dict’
DOOM Registered package ’company-prescient’
DOOM Registered package ’company-box’
DOOM Registered package ’helm’
DOOM Registered package ’helm-ag’
DOOM Registered package ’helm-c-yasnippet’
DOOM Registered package ’helm-company’
DOOM Registered package ’helm-describe-modes’ with recipe (:fetcher github :repo emacs-helm/helm-describe-modes)
DOOM Registered package ’helm-projectile’
DOOM Registered package ’swiper-helm’
DOOM Registered package ’helm-flx’
DOOM Registered package ’posframe’
DOOM Registered package ’doom-themes’
DOOM Registered package ’solaire-mode’
DOOM Registered package ’doom-modeline’
DOOM Registered package ’anzu’
DOOM Registered package ’evil-anzu’
DOOM Registered package ’evil-goggles’
DOOM Registered package ’fill-column-indicator’
DOOM Registered package ’hl-todo’
DOOM Registered package ’nav-flash’
DOOM Registered package ’treemacs’
DOOM Registered package ’treemacs-evil’
DOOM Registered package ’treemacs-projectile’
DOOM Registered package ’unicode-fonts’
DOOM Registered package ’git-gutter-fringe’
DOOM Registered package ’vi-tilde-fringe’
DOOM Registered package ’switch-window’
DOOM Registered package ’ace-window’
DOOM Registered package ’format-all’
DOOM Registered package ’evil-multiedit’
DOOM Registered package ’evil-mc’
DOOM Registered package ’multiple-cursors’
DOOM Registered package ’rotate-text’ with recipe (:fetcher github :repo debug-ito/rotate-text.el)
DOOM Registered package ’evil-vimish-fold’
DOOM Registered package ’dired-k’
DOOM Registered package ’ranger’
DOOM Registered package ’all-the-icons-dired’
DOOM Registered package ’imenu-anywhere’
DOOM Registered package ’imenu-list’
DOOM Registered package ’git-link’
DOOM Registered package ’git-timemachine’
DOOM Registered package ’gitconfig-mode’
DOOM Registered package ’gitignore-mode’
DOOM Registered package ’flyspell-correct’
DOOM Registered package ’flyspell-correct-ivy’
DOOM Registered package ’flyspell-correct-helm’
DOOM Registered package ’flyspell-correct-popup’
DOOM Registered package ’flycheck’
DOOM Registered package ’flycheck-popup-tip’
DOOM Registered package ’flycheck-posframe’
DOOM Registered package ’editorconfig’
DOOM Registered package ’makefile-executor’
DOOM Registered package ’magit’
DOOM Registered package ’forge’
DOOM Registered package ’magit-gitflow’
DOOM Registered package ’magit-todos’
DOOM Registered package ’evil-magit’
DOOM Registered package ’cmake-mode’
DOOM Registered package ’cuda-mode’
DOOM Registered package ’demangle-mode’
DOOM Registered package ’disaster’
DOOM Registered package ’modern-cpp-font-lock’
DOOM Registered package ’opencl-mode’
DOOM Registered package ’glsl-mode’
DOOM Registered package ’company-glsl’ with recipe (:fetcher github :repo Kaali/company-glsl)
DOOM Registered package ’ccls’
DOOM Registered package ’irony’
DOOM Registered package ’irony-eldoc’
DOOM Registered package ’flycheck-irony’
DOOM Registered package ’company-irony’
DOOM Registered package ’company-irony-c-headers’
DOOM Registered package ’rtags’
DOOM Registered package ’ivy-rtags’
DOOM Registered package ’helm-rtags’
  > (:lang cc) Couldn’t find glslangValidator. GLSL code completion
    is disabled
DOOM Registered package ’graphql-mode’
DOOM Registered package ’json-mode’
DOOM Registered package ’toml-mode’
DOOM Registered package ’yaml-mode’
DOOM Registered package ’csv-mode’
DOOM Registered package ’dhall-mode’
DOOM Registered package ’auto-compile’
DOOM Registered package ’highlight-quoted’
DOOM Registered package ’macrostep’
DOOM Registered package ’overseer’
DOOM Registered package ’elisp-def’
DOOM Registered package ’flycheck-cask’
DOOM Registered package ’coffee-mode’
DOOM Registered package ’js2-mode’
DOOM Registered package ’rjsx-mode’
DOOM Registered package ’typescript-mode’
DOOM Registered package ’eslintd-fix’
DOOM Registered package ’js2-refactor’
DOOM Registered package ’nodejs-repl’
DOOM Registered package ’npm-mode’
DOOM Registered package ’skewer-mode’
DOOM Registered package ’xref-js2’
DOOM Registered package ’tide’
DOOM Registered package ’markdown-mode’
DOOM Registered package ’markdown-toc’
DOOM Registered package ’pandoc-mode’
DOOM Registered package ’org-plus-contrib’
DOOM Registered package ’org’
DOOM Registered package ’org-bullets’ with recipe (:fetcher github :repo Kaligule/org-bullets)
DOOM Registered package ’org-yt’ with recipe (:fetcher github :repo TobiasZawada/org-yt)
DOOM Registered package ’toc-org’
DOOM Registered package ’evil-org’
DOOM Registered package ’org-pdfview’
DOOM Registered package ’org-download’
DOOM Registered package ’ob-async’
DOOM Registered package ’ob-mongo’
DOOM Registered package ’ob-sql-mode’
DOOM Registered package ’ob-translate’
DOOM Registered package ’ob-ipython’
DOOM Registered package ’ob-crystal’
DOOM Registered package ’ob-go’
DOOM Registered package ’ob-nim’
DOOM Registered package ’ob-racket’ with recipe (:fetcher github :repo DEADB17/ob-racket)
DOOM Registered package ’ob-restclient’
DOOM Registered package ’ob-rust’
DOOM Registered package ’ox-pandoc’
DOOM Registered package ’htmlize’
DOOM Registered package ’centered-window’ with recipe (:fetcher github :repo anler/centered-window-mode)
DOOM Registered package ’org-tree-slide’
DOOM Registered package ’ox-reveal’
DOOM Registered package ’nose’
DOOM Registered package ’python-pytest’
DOOM Registered package ’pip-requirements’
DOOM Registered package ’pipenv’
DOOM Registered package ’pyvenv’
DOOM Registered package ’pyenv-mode’
DOOM Registered package ’conda’
DOOM Registered package ’anaconda-mode’
DOOM Registered package ’company-anaconda’
DOOM Registered package ’rust-mode’
DOOM Registered package ’flycheck-rust’
DOOM Registered package ’racer’
  > (:lang rust) Syntax error: (error Key sequence M-SPC h m starts
    with non-prefix key M-SPC)
DOOM Registered package ’sbt-mode’
DOOM Registered package ’scala-mode’
DOOM Registered package ’lsp-scala’
DOOM Registered package ’ensime’
DOOM Registered package ’company-shell’
DOOM Registered package ’fish-mode’
DOOM Registered package ’circe’
DOOM Registered package ’circe-notifications’

```
</details>

**Comment by hlissner** [comment]
As of 8e866bb this should be resolved. Please update and let me know if that isn't the case!

**Comment by cameronrmorris** [comment]
Hi, looks like the change introduced in 8e866bb37f3bff6e6e8f0e8f2489ddb32e7dfb3c has broken some which-key previews:
![image](https://user-images.githubusercontent.com/636871/54087792-31d2d200-432d-11e9-857d-033822ba8d22.png)
Might be a separate issue

**Comment by hlissner** [comment]
@cameronrmorris the label issue should be fixed by dadcd29.

**Comment by cameronrmorris** [comment]
> @cameronrmorris the label issue should be fixed by [dadcd29](https://github.com/hlissner/doom-emacs/commit/dadcd29f279d161e566469d2be9f8ca691d5d747).

Yes it is fixed now :+1:

**Comment by shanavas786** [comment]
:+1:
